deploifai/training_server/create.py

```diff
@@ -0,0 +1,382 @@
+                        "name": "{}".format(info["name"]),
+                        "value": info
+                    }
+                    for info in datastorage_info
+                ],
+            }
+        )
+        if choose_datastorage == {}:
+            raise click.Abort()
+        pick_datastorage = choose_datastorage["datastorage"]
+    use_datastorage = True
+    if pick_datastorage["name"] == "no_datastorage":
+        use_datastorage = False
+    context.debug_msg("use_datastorage: {}".format(use_datastorage))
+    if use_datastorage:
+        # extract all the datastorage information required
+        datastorage = pick_datastorage
+        datastorage_id = datastorage["id"]
+
+    # extracting cloud profile information
+    cloud_profiles = context.api.get_cloud_profiles(workspace=workspace_name)
+    if len(cloud_profiles) == 0:
+        click.secho("Please create a cloud profile with")
+        click.secho("deploifai cloud-profile create", fg="yellow")
+        raise click.Abort()
+    if cloud_profile_name:
+        cloud_profile = False
+        for profile in cloud_profiles:
+            if profile.name == cloud_profile_name:
+                cloud_profile = profile
+                context.debug_msg("cloud profile info: {}".format(cloud_profile))
+        if not cloud_profile:
+            click.secho("Please provide a valid cloud profile name", fg="red")
+            raise click.Abort()
+    else:
+        choose_cloud_profile = prompt(
+            {
+                "type": "list",
+                "name": "cloud_profile",
+                "message": "Choose a cloud profile for training server",
+                "choices": [
+                    {
+                        "name": "{name}({workspace}) - {provider}".format(
+                            name=cloud_profile.name,
+                            workspace=cloud_profile.workspace,
+                            provider=cloud_profile.provider,
+                        ),
+                        "value": cloud_profile,
+                    }
+                    for cloud_profile in cloud_profiles
+                ],
+            }
+        )
+        if choose_cloud_profile == {}:
+            raise click.Abort()
+        cloud_profile = choose_cloud_profile["cloud_profile"]
+
+    # extracting the cloud profile information
+    cloud_profile_id = cloud_profile.id
+    cloud_profile_provider = cloud_profile.provider
+
+    # Obtaining information of python and framework version if using framework
+
+    if not framework:
+        questions = [
+            {
+                "type": "list",
+                "name": "ML Framework",
+                "message": "Which ML Framework will you be using?",
+                "choices": framework_options
+            },
+        ]
+        version = prompt(questions=questions)
+        if version == {}:
+            raise click.Abort()
+        framework = version["ML Framework"]
+
+    if framework != "no_framework":
```

Comment:
Shouldn't be necessary anymore with `--no-framework` option

Reply:
refactored